src/rgw/rgw_rest_pubsub.cc

```diff
@@ -86,18 +88,16 @@ int verify_topic_owner_or_policy(req_state* const s,
                                  const rgw_pubsub_topic& topic,
                                  const std::string& zonegroup_name,
                                  const uint64_t op) {
-  if (topic.user == s->owner.id) {
+  if (topic.user == s->owner.id || topic.user.empty()) {
```

Comment:
`topic.user.empty()` check will break the requirement for backward users that do not have user set but have set the policy explicitly, the policy will still not be verified.

Reply:
i don't think this can happen. if you set the policy, you will also set the user, only squid has the option for setting policy.

Reply:
> i don't think this can happen. if you set the policy, you will also set the user, only squid has the option for setting policy.

@yuvalif you can set the policy via SetAttributes and SetAttributes doesn't set the user id. So if someone has legacy topics and upgraded to squid  can still set policy n have user empty

Reply:
ok. will move that check into the "no policy" section